### PR TITLE
Kiali docs for GA.

### PR DESCRIPTION
**Pull request by JStickler**
Overview Kiali docs for GA.
@jdoyleoss, @brianredbeard, @mwringe

### CODE REVIEW

**Comment by vikram-redhat** [comment]
Preview build is here: https://kiali--ocpdocs.netlify.com/openshift-enterprise/latest/service_mesh/service_mesh_install/ossm-kiali.html

**Comment by JStickler** [comment]
@gbaufake

File: modules/ossm-kiali-features.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+* *Topology* – Visualize how your applications, services, or workloads communicate via the Kiali graph.
+
+* *Metrics* – Predefined metrics dashboards let you chart service mesh and application performance for Go, Node.js. Quarkus, Spring Boot, Thorntail and Vert.x.  You can also create your own custom dashboards.
+
+* *Tracing* – Integration with Jaeger lets you follow the path of a request through various microservices that make up an application.
+
+* *Validations* – Perform advanced validations on the most common Istio Objects (Destination Rules, Service Entries, Virtual Services, and so on).
+
+* *Configuration* – Optional ability to create, update and delete Istio routing configuration via wizards in the Kiali Console.
```
Comment (hhovsepy) [inline modules/ossm-kiali-features.adoc]: It also allows to update and delete Istio routing configuration directly in editor page outside of wizard.
"... via wizards or directly in editor page in the Kiali ...."

File: modules/ossm-kiali-architecture.adoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+This CONCEPT module included in the following assemblies:
+-ossm-kiali.adoc
+////
+
+[id="ossm-kiali-architecture_{context}"]
+= Kiali architecture
+
+Kiali is composed of two components: the Kiali application and the Kiali console.
+
+* *Kiali application* (back end) – This component runs in the container application platform and communicates with the service mesh components, retrieves and processes data, and exposes this data to the console. The Kiali application does not need storage. When you are running the application locally, all configurations are set in files and environment variables (including the user and password used for authentication). When deploying the application to a cluster, configurations are set in configmaps and secrets.
```
Comment (hhovsepy) [inline modules/ossm-kiali-architecture.adoc]: Is the local run of the application supported way?

**Comment by mwringe** [comment]
Is there a section on how to configure Kiali through the ServiceMeshControlPlane CR?

We might also want to mention that the kiali CR is installed as a restricted resource, only the Maistra Operator or cluster admin users should be able to create or modify a kiali CR directly. Cluster admins should not grant permission to other users to create or modify Kiali CR as it can be used to escalate their privileges.

File: modules/ossm-kiali-architecture.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+* *Prometheus* - A dedicated Prometheus instance is included as part of the {ProductName} installation.  When Istio telemetry is enabled, metrics data is stored in Prometheus. Kiali uses this Prometheus data to determine the mesh topology, display metrics, calculate health, show possible problems, etc. Kiali communicates directly with Prometheus and assumes the data schema used by Istio Telemetery. Prometheus is an Istio dependency and a hard dependency for Kiali, and many of Kiali’s features will not work without Prometheus.
+
+* *Cluster API* - Kiali uses the API of the OpenShift container application platform (cluster API) in order to fetch and resolve service mesh configurations. Kiali queries the cluster API to retrieve, for example, definitions for: namespaces, services, deployments, pods, and other entities. Kiali also makes queries to resolve relationships between the different cluster entities. The cluster API is also queried to retrieve Istio configurations like: virtual services, destination rules, route rules, gateways, quotas, etc.
```
Comment (bergerhoffer) [inline modules/ossm-kiali-architecture.adoc]: * Apologies if this is just my ignorance, but is "OpenShift container application platform" actually different than "OpenShift Container Platform"? If not, then you should use the {product-title} attribute for OCP in our document attribtues file.
* I don't think the colon after "for" or "like" is necessary.

File: modules/ossm-kiali-service-mesh.adoc [hunk 4]
```diff
@@ -0,0 +1,15 @@
+////
+This CONCEPT module included in the following assemblies:
+-assembly.adoc
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-service-mesh.adoc]: Yeah, I just wasn't sure where when I wrote it.  =)

File: modules/ossm-kiali-architecture.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+* *Jaeger* - Jaeger is optional, but is installed by default as part of the {ProductName} installation. When you install Jaeger as part of the default {ProductName} installation, the Kiali console includes a tab to display Jaeger’s tracing data. Note that tracing data will not be available if you disable Istio’s distributed tracing feature.
+
+* *Grafana* - Grafana is optional, but is installed by default as part of the {ProductName} installation. When available, the metrics pages of Kiali 	display links to direct the user to the same metric in Grafana. To enable this feature, make sure Kiali is properly configured for Grafana integration.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-architecture.adoc]: Deleted that line.

File: modules/ossm-kiali-architecture.adoc [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+* *Cluster API* - Kiali uses the API of the OpenShift container application platform (cluster API) in order to fetch and resolve service mesh configurations. Kiali queries the cluster API to retrieve, for example, definitions for: namespaces, services, deployments, pods, and other entities. Kiali also makes queries to resolve relationships between the different cluster entities. The cluster API is also queried to retrieve Istio configurations like: virtual services, destination rules, route rules, gateways, quotas, etc.
+
+* *Jaeger* - Jaeger is optional, but is installed by default as part of the {ProductName} installation. When you install Jaeger as part of the default {ProductName} installation, the Kiali console includes a tab to display Jaeger’s tracing data. Note that tracing data will not be available if you disable Istio’s distributed tracing feature.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-architecture.adoc]: Added notes.

File: modules/ossm-kiali-overview.adoc [hunk 7]
```diff
@@ -0,0 +1,13 @@
+////
+This CONCEPT module included in the following assemblies:
+-ossm-kiali.adoc
+////
+
+[id="ossm-kiali-overview_{context}"]
+= Kiali overview
+
+Kiali works with {ProductName} to provide application observability for OpenShift. Kiali helps you define, validate, and observe your Istio service mesh. It helps you to understand the structure of your service mesh by inferring the topology, and also provides information about the health of your mesh.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-overview.adoc]: Rewrote.

File: modules/ossm-kiali-service-mesh.adoc [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+////
+
+[id="ossm-kiali-service-mesh_{context}"]
+= Kiali and service mesh
+
+Installing Kiali  via the Service Mesh on OpenShift differs from community Kiali installations in multiple ways.  These modifications are sometimes necessary to resolve issues, provide additional features, or to handle differences when deploying on OpenShift or OKD.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-service-mesh.adoc]: Added that configuration information.

File: modules/ossm-kiali-architecture.adoc (hunk 3 above)
Comment (JStickler) [reply modules/ossm-kiali-architecture.adoc]: No, that should be OpenShift Container Platform.

File: modules/ossm-kiali-overview.adoc [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+Kiali provides an interactive graph view of your namespace in real time that provides visibility into features like circuit breakers, request rates, latency, and even graphs of traffic flows. Kiali offers insights about components at different levels, from Applications to Services and Workloads, and can display the interactions with contextual information and charts on the selected graph node or edge. Kiali also provides the ability to validate your Istio configurations, such as gateways, destination rules, VirtualServices, mesh policies, and more. Kiali provides detailed metrics, and a basic Grafana integration is available for advanced queries. Distributed tracing is provided by integrating Jaeger into the Kiali console.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-overview.adoc]: I left in the capitalization for "Applications to Services and Workloads" because those are tabs and also viewing options (so to match the UI).

File: modules/ossm-kiali-service-mesh.adoc [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+* Kiali has been enabled by default.
+* Ingress has been enabled by default.
+* Updates have been made to the Kiali ConfigMap.
+* Updates have been made to the ClusterRole settings for Kiali.
+* A demo Secret has been added that will get 	created using values from kiali.dashboard.user and kiali.dashboard.passphrase, if specified.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-service-mesh.adoc]: That line was deleted due to comments from Matt.

[the remark this answers is not in the document]
Comment (brian-avery) [reply modules/ossm-kiali-service-mesh.adoc]: Correct.

File: modules/ossm-kiali-architecture.adoc (hunk 2 above)
Comment (hhovsepy) [reply modules/ossm-kiali-architecture.adoc]: verified

File: modules/ossm-kiali-features.adoc (hunk 1 above)
Comment (hhovsepy) [reply modules/ossm-kiali-features.adoc]: verified

File: modules/ossm-kiali-service-mesh.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-service-mesh.adoc]: Updated and then moved the content from here to where we are documenting the SMCP configurations in Elise's PR (https://github.com/openshift/openshift-docs/pull/16326 ).

**Comment by JStickler** [comment]
Looks like my build is failing due to problems in a file that is not part of this PR.  I need to track down the fix and rebase, but since it's a holiday weekend, that might take a couple of days.

File: modules/ossm-kiali-architecture.adoc [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+////
+
+[id="ossm-kiali-architecture_{context}"]
+= Kiali architecture
+
+Kiali is composed of two components: the Kiali application and the Kiali console.
+
+* *Kiali application* (back end) – This component runs in the container application platform and communicates with the service mesh components, retrieves and processes data, and exposes this data to the console. The Kiali application does not need storage. When deploying the application to a cluster, configurations are set in ConfigMaps and secrets.
+
+* *Kiali console* (front end) – The Kiali console is a web application. 	The Kiali application serves the Kiali console, which then queries the application for data in order to present it to the user.
```
Comment (abonas) [inline modules/ossm-kiali-architecture.adoc]: this sentence is really hard to follow and understand. It sounds like it's a loop. "The Kiali application serves the Kiali console which then queries the application for data...."

File: modules/ossm-kiali-architecture.adoc [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+This CONCEPT module included in the following assemblies:
+-ossm-kiali.adoc
+////
+
+[id="ossm-kiali-architecture_{context}"]
+= Kiali architecture
+
+Kiali is composed of two components: the Kiali application and the Kiali console.
+
+* *Kiali application* (back end) – This component runs in the container application platform and communicates with the service mesh components, retrieves and processes data, and exposes this data to the console. The Kiali application does not need storage. When deploying the application to a cluster, configurations are set in ConfigMaps and secrets.
```
Comment (abonas) [inline modules/ossm-kiali-architecture.adoc]: Why not simply call the backend as the backend? An application is comprised of the UI/console + backend.

File: modules/ossm-kiali-architecture.adoc [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+In addition, Kiali depends on external services and components provided by the container application platform and Istio.
+
+* *Red Hat Service Mesh* (Istio) - Istio is a Kiali requirement. Istio is the component that provides and controls the service mesh. Although Kiali and Istio can be installed separately, Kiali depends on Istio and will not work if it is not present. Kiali needs to retrieve Istio data and configurations, which are exposed through Prometheus and the cluster API.
+
+* *Prometheus* - A dedicated Prometheus instance is included as part of the {ProductName} installation.  When Istio telemetry is enabled, metrics data is stored in Prometheus. Kiali uses this Prometheus data to determine the mesh topology, display metrics, calculate health, show possible problems, and so on. Kiali communicates directly with Prometheus and assumes the data schema used by Istio Telemetery. Prometheus is an Istio dependency and a hard dependency for Kiali, and many of Kiali’s features will not work without Prometheus.
```
Comment (abonas) [inline modules/ossm-kiali-architecture.adoc]: is the "ProductName" here equivalent to the "product-title" on line 21?

File: modules/ossm-kiali-architecture.adoc [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+* *Cluster API* - Kiali uses the API of the {product-title} (cluster API) in order to fetch and resolve service mesh configurations. Kiali queries the cluster API to retrieve, for example, definitions for namespaces, services, deployments, pods, and other entities. Kiali also makes queries to resolve relationships between the different cluster entities. The cluster API is also queried to retrieve Istio configurations like virtual services, destination rules, route rules, gateways, quotas, and so on.
+
```
Comment (abonas) [inline modules/ossm-kiali-architecture.adoc]: @lucasponce @mwringe is indeed all Istio config retrieved via the cluster api? Nothing is retrieved via Istio api?

File: modules/ossm-kiali-architecture.adoc [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+* *Jaeger* - Jaeger is optional, but is installed by default as part of the {ProductName} installation. When you install Jaeger as part of the default {ProductName} installation, the Kiali console includes a tab to display Jaeger’s tracing data. Note that tracing data will not be available if you disable Istio’s distributed tracing feature.  Also note that user must have access to the namespace where the control plane is installed in order to view Jaeger data.
+
+* *Grafana* - Grafana is optional, but is installed by default as part of the {ProductName} installation. When available, the metrics pages of Kiali 	display links to direct the user to the same metric in Grafana. Note that user must have access to the namespace where the control plane is installed in order to view links to the Grafana dashboard and view Grafana data.
```
Comment (abonas) [inline modules/ossm-kiali-architecture.adoc]: it looks like there's a very big blank space between "Kiali" and "display links" but it could be a presentation issue on my side.

File: modules/ossm-kiali-service-mesh.adoc [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+This CONCEPT module included in the following assemblies:
+-ossm-kiali.adoc
+////
+
+[id="ossm-kiali-service-mesh_{context}"]
+= Kiali and service mesh
+
+Installing Kiali  via the Service Mesh on {product-title} differs from community Kiali installations in multiple ways.  These modifications are sometimes necessary to resolve issues, provide additional features, or to handle differences when deploying on {product-title}.
+
+* Kiali has been enabled by default.
```
Comment (abonas) [inline modules/ossm-kiali-service-mesh.adoc]: How does this section relate to the paragraph above it?

File: modules/ossm-kiali-architecture.adoc (hunk 14 above)
Comment (lucasponce) [reply modules/ossm-kiali-architecture.adoc]: Yes. Istio doesn't not expose a service that Kiali uses.
All information is fetched via Cluster API using Istio CRDs.
So, in the this context the expression is correct.

File: modules/ossm-kiali-architecture.adoc (hunk 15 above)
Comment (abonas) [reply modules/ossm-kiali-architecture.adoc]: @jotak can you please review this part ^^

Comment (jotak) [reply modules/ossm-kiali-architecture.adoc]: Looks good to me!

File: modules/ossm-kiali-architecture.adoc (hunk 13 above)
Comment (JStickler) [reply modules/ossm-kiali-architecture.adoc]: ProductName = Service Mesh 
Product-Title = OpenShift

File: modules/ossm-kiali-architecture.adoc (hunk 11 above)
Comment (JStickler) [reply modules/ossm-kiali-architecture.adoc]: Revised

File: modules/ossm-kiali-architecture.adoc [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+
+* *Cluster API* - Kiali uses the API of the {product-title} (cluster API) in order to fetch and resolve service mesh configurations. Kiali queries the cluster API to retrieve, for example, definitions for namespaces, services, deployments, pods, and other entities. Kiali also makes queries to resolve relationships between the different cluster entities. The cluster API is also queried to retrieve Istio configurations like virtual services, destination rules, route rules, gateways, quotas, and so on.
+
+* *Jaeger* - Jaeger is optional, but is installed by default as part of the {ProductName} installation. When you install Jaeger as part of the default {ProductName} installation, the Kiali console includes a tab to display Jaeger’s tracing data. Note that tracing data will not be available if you disable Istio’s distributed tracing feature.  Also note that user must have access to the namespace where the control plane is installed in order to view Jaeger data.
```
Comment (travisrogers05) [inline modules/ossm-kiali-architecture.adoc]: > Also note that user must have access to the namespace where the control plane is installed in order to view Jaeger data.

Is this view only permissions that are required?  Should we specify the exact permissions?
